lib/logic.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class TicTacLogic
+  attr_accessor :player1, :player2
+  attr_reader :board, :play, :playert
+
+  def initialize
+    @play = true
+    @board = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    @playert = true
+  end
+
+  def move(num)
+    if @board[num - 1].is_a?(Integer) && num >= 0 && num <= 9
+      @board[num - 1] = @playert ? 'X' : 'O'
+      return true
+    end
+    false
+  end
+
+  def game_end?
+    if victory || vic_diagonal
+      @play = false
+      "Game over - Winner: #{@playert ? @player1 : @player2}"
+    elsif @board.none?(Integer)
+      @play = false
+      "Game over - it's a tie"
+    else
+      @playert = !@playert
+      "Please #{@playert ? @player1 : @player2}, make a move"
+    end
+  end
```

Comment:
Returning string is not a good habit
Please, return a value and use that value to decided what to print